store/src/main/java/com/dropbox/android/external/store4/impl/FetcherController.kt

```diff
@@ -64,10 +65,16 @@ internal class FetcherController<Key, Input, Output>(
                 scope = scope,
                 bufferSize = 0,
                 source = flow { emitAll(realFetcher(key)) }.map {
-                    StoreResponse.Data(
-                        it,
-                        origin = ResponseOrigin.Fetcher
-                    ) as StoreResponse<Input>
+                    when (it) {
+                        is FetcherResult.Data -> StoreResponse.Data(
+                            it.value,
+                            origin = ResponseOrigin.Fetcher
+                        ) as StoreResponse<Input>
+                        is FetcherResult.Error -> StoreResponse.Error(
+                            it.error,
+                            origin = ResponseOrigin.Fetcher
+                        )
+                    }
                 }.catch {
```

Comment:
I think in this new world, it should crash the whole flow and hopefully the app

Reply:
This change is only intended to allow the use of a wider variety of fetchers. Why do you think we should change the behavior of Store? In general we don't propagate exceptions but concert then to errors, why would this change affect this?

Reply:
because in the previous API, they can only return values hence we have to catch exceptions for errors.

In this new world, they get the ability to return errors so any exception can be considered exceptional. FYI this is also what paging does but we are considering options to maybe catch IO Exceptions by default to be more pragmatic. 

A reason to opt into this new API would be to catch errors like NPE's and if we eat them, it would probably make people unhappy.